app/src/main/java/org/schabi/newpipe/settings/NotificationsSettingsFragment.kt

```diff
@@ -0,0 +1,108 @@
+import org.schabi.newpipe.notifications.NotificationWorker
+import org.schabi.newpipe.notifications.ScheduleOptions
+import java.util.*
+
+class NotificationsSettingsFragment : BasePreferenceFragment(), OnSharedPreferenceChangeListener {
+
+    private var notificationWarningSnackbar: Snackbar? = null
+    private var loader: Disposable? = null
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        addPreferencesFromResource(R.xml.notifications_settings)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        defaultPreferences.registerOnSharedPreferenceChangeListener(this)
+    }
+
+    override fun onStop() {
+        defaultPreferences.unregisterOnSharedPreferenceChangeListener(this)
+        super.onStop()
+    }
+
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
+        val ctx = context ?: return
+        if (key == getString(R.string.streams_notifications_interval_key) || key == getString(R.string.streams_notifications_network_key)) {
+            NotificationWorker.schedule(ctx, ScheduleOptions.from(ctx), true)
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        val enabled = NotificationHelper.isNotificationsEnabledNative(context)
+        preferenceScreen.isEnabled = enabled
+        if (!enabled) {
+            if (notificationWarningSnackbar == null) {
+                notificationWarningSnackbar = Snackbar.make(
+                        listView,
+                        R.string.notifications_disabled,
+                        Snackbar.LENGTH_INDEFINITE
+                ).apply {
+                    setAction(R.string.settings) { v ->
+                        NotificationHelper.openNativeSettingsScreen(v.context)
+                    }
+                    setActionTextColor(ContextCompat.getColor(requireContext(),
+                            R.color.snackbar_action_color))
+                    addCallback(object : Snackbar.Callback() {
+                        override fun onDismissed(transientBottomBar: Snackbar, event: Int) {
+                            super.onDismissed(transientBottomBar, event)
+                            notificationWarningSnackbar = null
+                        }
+                    })
+                    show()
+                }
+            }
+        } else {
+            notificationWarningSnackbar?.dismiss()
+            notificationWarningSnackbar = null
+        }
+        loader?.dispose()
+        loader = SubscriptionManager(requireContext())
+                .subscriptions()
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::updateSubscriptions, this::onError)
+    }
+
+    override fun onPause() {
+        loader?.dispose()
+        loader = null
+        super.onPause()
+    }
+
+    private fun updateSubscriptions(list: List<SubscriptionEntity>) {
+        var notified = 0
+        for (o in list) {
```

Comment:
Instead of calling it `o` and `list`, I think `subscription` and `subscriptions` would be better names